app/crud.py

```diff
@@ -0,0 +1,95 @@
+from datetime import datetime
+
+from fastapi import HTTPException
+from sqlalchemy.future import select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.models import DBCity, DBTemperature
+from app.schemas import CityCreate
+from app.utills import fetch_temperature
+
+
+async def get_all_cities(db: AsyncSession, skip: int = 0, limit: int = 100):
+    result = await db.execute(select(DBCity).offset(skip).limit(limit))
+    return result.scalars().all()
+
+
+async def create_new_city(db: AsyncSession, city: CityCreate):
+    db_city = DBCity(name=city.name, additional_info=city.additional_info)
+    db.add(db_city)
```

Comment:
In your `create_new_city` function, you are directly committing the transaction after adding a new city. This may not be the best practice as it can lead to performance issues. It's often better to use a unit of work pattern where you make all the database changes you need to make within a single session and then commit them all at once at the end.